packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -158,7 +160,14 @@ export const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
     return isQueryHistoryAutocompleteEnabled
       ? createQueryWithHistoryAutocompleter(
           savedQueries
-            .filter((query) => !('update' in query))
+            .filter((query) => {
+              const isOptionNameInQuery =
+                optionName !== 'filter'
+                  ? optionName in query.queryProperties
+                  : true;
```

Comment:
Another way to write this is
```suggestion
              const isOptionNameInQuery = optionName === 'filter' || optionName in query.queryProperties;
```

Reply:
Ah oops, this probably makes the reformat check fail, need to run reformat on it.